internal/conf/configuration.go

```diff
@@ -417,6 +419,21 @@ func (config *GlobalConfiguration) ApplyDefaults() error {
 		config.External.AllowedIdTokenIssuers = append(config.External.AllowedIdTokenIssuers, "https://appleid.apple.com", "https://accounts.google.com")
 	}
 
+	if config.API.DomainAllowList == nil && config.API.ExternalURL != "" {
+		// for backward compatibility, always include the ExternalURL in the DomainAllowList
+		config.API.DomainAllowList = []string{config.API.ExternalURL}
+	}
+
+	// Get the hostname and create a map of each hostname to URL
+	config.API.DomainAllowListMap = make(map[string]*url.URL)
```

Comment:
We store an allow list for the allowed domains and then process it to map the hostname to each URL. This is useful for when we need to validate the hostname in the request and return the entire URL.

Reply:
Why is this necessary? Usually you can't just pick your own `Host` header in HTTPS. It has to match the hostname of the TLS cert, so putting another layer of "validation" doesn't do much yet it complicates the config quite a bit.

Reply:
Hmm good point, i can remove this and just use the hostname from the request URL instead. Anyway, if the custom domain used isn't being setup correctly, then the request won't even reach gotrue.

Reply:
Yup exactly.